src/controller/tests/TestAutoCommissioner.cpp

```diff
@@ -685,7 +781,281 @@ TEST_F(AutoCommissionerTest, IsSecondaryNetworkSupportedCombinations)
+                                         5678,                                  // peerSessionId
+                                         1,                                     // fabric index
+                                         GetDefaultMRPConfig()                  // MRP config
+    );
+    SessionHandle sessionHandle(testSession);
+    OperationalDeviceProxy operationalProxy(&exchangeMgr, sessionHandle);
+
+    privateConfigCommissioner.SetOperationalDeviceProxy(operationalProxy);
+
+    chip::Controller::ReadCommissioningInfo & commissioningInfo = privateConfigCommissioner.GetDeviceCommissioningInfo();
+    commissioningInfo.requiresUTC                               = true;
+    commissioningInfo.requiresTimeZone                          = true;
+    commissioningInfo.maxTimeZoneSize                           = 128;
+    commissioningInfo.remoteNodeId                              = 0xDEADBEEF;
+    commissioningInfo.supportsConcurrentConnection              = false;
+    commissioningInfo.network.wifi.endpoint                     = kRootEndpointId;
+    commissioningInfo.network.thread.endpoint                   = kRootEndpointId;
+
+    privateConfigCommissioner.SetNeedsDST();
+    privateConfigCommissioner.TrySecondaryNetwork();
+
+    privateConfigCommissioner.CleanupCommissioning();
+    // Default parameters are expected since CleanupCommissioning resets everything,
+    // including internal structs such as OperationalDeviceProxy and DeviceCommissioningInfo.
+    EXPECT_EQ(privateConfigCommissioner.TryingSecondaryNetwork(), false);
+    EXPECT_TRUE(privateConfigCommissioner.GetPAI().empty());
+    EXPECT_TRUE(privateConfigCommissioner.GetDAC().empty());
+    EXPECT_EQ(privateConfigCommissioner.GetCommissioneeDeviceProxy(), nullptr);
+    test_helpers::ExpectOperationalDeviceProxyEq(privateConfigCommissioner.GetOperationalDeviceProxy());
+    test_helpers::ExpectReadCommissioningInfoEq(privateConfigCommissioner.GetDeviceCommissioningInfo());
+    EXPECT_EQ(privateConfigCommissioner.GetNeedsDST(), false);
+}
+
+TEST_F(AutoCommissionerTest, GetCommandTimeout_CalculatesCorrectlyWithoutActiveSession)
+{
+
+    AutoCommissionerTestAccess privateConfigCommissioner(&mCommissioner);
+    Controller::ReadCommissioningInfo & commissioningInfo = privateConfigCommissioner.GetDeviceCommissioningInfo();
+    CommissioneeDeviceProxy device;
+    privateConfigCommissioner.SetDeviceCommissioneeProxy(&device);
+    commissioningInfo.network.wifi.minConnectionTime = 20;
+    CommissioningStage stage                         = kWiFiNetworkEnable;
+    System::Clock::Timeout expectedTimeout           = kMinimumCommissioningStepTimeout;
+
+    EXPECT_EQ(privateConfigCommissioner.GetCommandTimeout(privateConfigCommissioner.GetCommissioneeDeviceProxy(), stage),
+              MakeOptional(expectedTimeout));
+}
+
+TEST_F(AutoCommissionerTest, GetCommandTimeout_CalculatesCorrectlyWithActiveSession)
```

Comment:
this doesn't appear to be checking anything related to the session. The timeout for thread network connection on an active session should be the min network enable time PLUS the round trip timeout, with the minimum set to the spec defined minimum timeout.